### PR TITLE
build: Allow passing in an existing Redis instance

### PR DESCRIPTION
**Pull request by Kikobeats**
useful to reuse the same redis connection!

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -10,3 +11,10 @@ keyvOfficialTests(test, Keyv, redisURI, 'redis://foo');
 
 const store = () => new KeyvRedis(redisURI);
 keyvTestSuite(test, Keyv, store);
+
+test('reuse a redis instance', t => {
+	const redis = new Redis(redisURI);
+	redis.foo = 'bar';
+	const keyv = new KeyvRedis(redis);
+  t.is(keyv.redis.foo, 'bar');
```
Comment (lukechilds) [inline test/test.js]: Maybe also do a quick set/get in here and check the return values so we know if anything breaks.

Comment (Kikobeats) [reply test/test.js]: done 🙂

**Comment by Kikobeats** [comment]
@lukechilds docs extended!

File: test/test.js [hunk 2]
```diff
@@ -10,3 +11,13 @@ keyvOfficialTests(test, Keyv, redisURI, 'redis://foo');
...
 keyvTestSuite(test, Keyv, store);
+
+test('reuse a redis instance', async t => {
+	const redis = new Redis(redisURI);
+	redis.foo = 'bar';
+	const keyv = new KeyvRedis(redis);
+  t.is(keyv.redis.foo, 'bar');
+
+  await keyv.set('foo', 'bar');
+        t.true(await redis.get('foo') === await keyv.get('foo'));
```
Comment (lukechilds) [inline test/test.js]: Looks like you have too much indentation here.

Also could you add another test for `keyv.get('foo') === 'bar`. Good idea to test `redis.get` against `keyv.get` but this test will pass if they are both incorrect (e.g both return undefined).

Comment (Kikobeats) [reply test/test.js]: done 👍 

I indented using `xo --fix`

**Comment by Kikobeats** [comment]
After merge and before release a new version tests will run on master so you'll be sure all is fine 🙂

**Comment by Kikobeats** [comment]
Screenshot 2019-11-17 at 10 01 53

**Comment by lukechilds** [comment]
Awesome, thanks!

**Pull request merged**